packages/@aws-cdk/integ-tests/lib/assertions/sdk.ts

```diff
@@ -49,6 +62,8 @@ export class AwsApiCall extends ApiCallBase {
     this.provider.addPolicyStatementFromSdkCall(props.service, props.api);
     this.name = `${props.service}${props.api}`;
 
+    const salt = this.node.tryGetContext(ASSET_SALT_CONTEXT_KEY);
+
```

Comment:
Why this? It's not what that key was intended for.

But I think I do see: is it because otherwise one cannot get a stable snapshot of the template?

If so, this needs @corymhall in here, who's currently the deepest into integ testing. The integ testing facilities should automatically be producing consistent values whenever they know their snapshot is going to be taken.

Reply:
yes your right... i thought i might need to use another way or use another key..
It is due to the snapshot test stability.

stems from the salt = Date.now()

perhaps if we could have some way to make code aware it was in integ-runner other than context. Something like jest.spyOn() to help make Date.now() static was another thought.

Reply:
I think for this I would rather make the runner aware of the assertions. On diff it could just ignore the salt (like we do for asset hashes).